src/Radicle/Ipfs.hs

```diff
@@ -163,12 +184,12 @@ subscribe topic messageHandler = runResourceT $ do
 -- | Publish a message to a topic.
 publish :: Text -> LByteString -> IO ()
 publish topic message =
-    void $ ipfsHttpPost' "pubsub/pub" [("arg", topic)] "data" message
+    catchHttp $ void $ ipfsHttpPost' "pubsub/pub" [("arg", topic)] "data" message
 
 newtype KeyGenResponse = KeyGenResponse IpnsId
 
 keyGen :: Text -> IO KeyGenResponse
-keyGen name = ipfsHttpGet "key/gen" [("arg", name), ("type", "ed25519")]
+keyGen name = catchHttp $ ipfsHttpGet "key/gen" [("arg", name), ("type", "ed25519")]
```

Comment:
Can’t we move `catchHttp` to `ipfsHttpGet` and `ipfsHttpPost`?

Reply:
:+1: